app/classification/views.py

```diff
@@ -172,29 +213,36 @@ def submit_classification():
                             contract_id_list=[]
                            )
 
-  # try:
-  #   # add classification to the database
-  #   db.session.add(submitdata)
-  #   db.session.commit()
-  #   print('You have successfully added a new classification.')
-  # except SQLAlchemyError as e:
-  #   # in case classification already exists
-  #   print(e)
+  try:
+    # add classification to the database
+    db.session.add(submitdata)
+    db.session.commit()
+    print('You have successfully added a new classification.')
+  except SQLAlchemyError as e:
+    # in case classification already exists
+    print(e)
 
 
 
 
   froblist = compare_submission(domainlist, geolist, sourcelist, usagelist, identruleslist)
   
   print(json.dumps(froblist, indent=4))
-  foundclass = froblist[0][1] == 0
   classdata = classificationdbo.query.get_or_404(froblist[0][0])
-  retdict = { "found" : str(foundclass),
-              "classid": classdata.classification_id,
-              "recipeid": classdata.recipe_id,
-              "label": classdata.label,
-              "identlist" : [ str(x[0]) + "|" +str(x[1]) for x in identruleslist ]
-            }
+  if froblist[0][1] == 0:
```

Comment:
Maybe refactor as:

```
retdict = { "found" : True if froblist[0][1] == 0 else False

```
to drop duplicate below, since it looks like only the value for "found" is changing here.

Reply:
I did think of that but something tripped me up on the jsonify step or in the Javascript handling of the return value, such that it wasn't recognizing something that python saw as a false value.  This way worked and I am not sure I can explain why.  Good catch though and definitely going to add a trello ticket for future.